src/utils.ts

```diff
@@ -182,25 +188,62 @@ export const bufferToBase64UrlEncoded = input => {
   );
 };
 
-const getJSON = async (url, options) => {
-  const response = await fetch(url, options);
+const fetchWithTimeout = (url, options, timeout = DEFAULT_FETCH_TIMEOUT_MS) => {
+  // The promise will resolve with one of these two promises (the fetch and the timeout), whichever completes first.
+  return Promise.race([
```

Comment:
`Promise.race` ... handy!

Reply:
I assume this ends the `fetch()` process so that's not being tried in the background while the app continues?